### PR TITLE
[delivery_*] packages management

### PR DESCRIPTION
**Pull request by mgeubelle**
https://www.odoo.com/web/#id=21195&view_type=form&model=project.task&action=333&active_id=248&menu_id=4720

### CODE REVIEW

File: addons/delivery/models/stock_picking.py [hunk 1]
```diff
@@ -21,6 +21,42 @@ def _compute_weight(self):
         self.weight = weight
 
     weight = fields.Float(compute='_compute_weight')
+    shipping_weight = fields.Float(string='Shipping Weight', help="Can be changed during the 'put in pack' to adjust the weight of the shipping.")
+
+    def save(self, cr, uid, ids, context=None):
```
Comment (rim-odoo) [inline addons/delivery/models/stock_picking.py]: @mge-odoo Old API ?

File: addons/delivery/models/stock_picking.py [hunk 2]
```diff
@@ -21,6 +21,42 @@ def _compute_weight(self):
...
+class StockPackOperation(models.Model):
+    _inherit = 'stock.pack.operation'
+
+    @api.multi
+    def manage_package_type(self):
+
+        delivery_type = self.picking_id.carrier_id.delivery_type
+        carrier_packaging_field = delivery_type and delivery_type not in ['fixed', 'base_on_rule'] and delivery_type + '_packaging_type' or 'name'
+
+        view = self.env['ir.model.data'].xmlid_to_res_id('delivery.view_quant_package_form_save')
```
Comment (rim-odoo) [inline addons/delivery/models/stock_picking.py]: note to myself: self.env.ref

File: addons/delivery/models/stock_picking.py [hunk 3]
```diff
@@ -21,6 +21,32 @@ def _compute_weight(self):
...
+
+
+class StockPackOperation(models.Model):
+    _inherit = 'stock.pack.operation'
+
+    @api.multi
+    def manage_package_type(self):
+
+        delivery_type = self.picking_id.carrier_id.delivery_type
+        carrier_packaging_field = delivery_type and delivery_type not in ['fixed', 'base_on_rule'] and delivery_type + '_packaging_type' or 'name'
```
Comment (antonylesuisse) [inline addons/delivery/models/stock_picking.py]: remove

File: addons/delivery/models/stock_picking.py [hunk 4]
```diff
@@ -21,6 +21,32 @@ def _compute_weight(self):
...
+            'name': _('Package Details'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'stock.quant.package',
+            'view_id': view,
+            'target': 'new',
+            'res_id': self.result_package_id.id,
+            'context': {
+                # Used to filter the packaging_id related to the carrier_id
+                'carrier_packaging_field': carrier_packaging_field
```
Comment (antonylesuisse) [inline addons/delivery/models/stock_picking.py]: remove

File: addons/delivery/models/stock_picking.py [hunk 5]
```diff
@@ -82,6 +116,34 @@ def do_transfer(self):
...
+    def put_in_pack(self):
+        package_id = super(StockPicking, self).put_in_pack()
+        package = self.env['stock.quant.package'].browse(package_id)
+
+        # By default, sum the weights of all package operations contained in this package
+        pack_operation_ids = self.env['stock.pack.operation'].search([('result_package_id', '=', package_id)])
+        package_weight = sum([x.qty_done * x.product_id.weight for x in pack_operation_ids])
+        package.shipping_weight = package_weight
+
+        delivery_type = self.carrier_id.delivery_type
```
Comment (antonylesuisse) [inline addons/delivery/models/stock_picking.py]: remove

File: addons/delivery/models/stock_picking.py [hunk 6]
```diff
@@ -82,6 +116,34 @@ def do_transfer(self):
...
+        carrier_packaging_field = delivery_type and delivery_type not in ['fixed', 'base_on_rule'] and delivery_type + '_packaging_type' or 'name'
```
Comment (antonylesuisse) [inline addons/delivery/models/stock_picking.py]: remove

**Pull request merged**